api/envoy/config/filter/http/transformation/v2/transformation_filter.proto

```diff
@@ -314,6 +314,12 @@ message TransformationTemplate {
   // rendered string values to be escaped in order to make valid JSON/YAML strings
   bool escape_characters = 12;
 
+  // The character to use as a delimiter when accessing dynamic metadata values
+  // in templates. Defaults to ':' if unset.
+  // See https://www.envoyproxy.io/docs/envoy/v1.31.1/configuration/observability/access_log/usage.html
+  // for more information.
+  string string_delimiter = 14;
```

Comment:
why is this configurable? can we just have it be an internal constant?

Reply:
I don't have a strong feeling about this, can always expose it later